network/requestTracker.go

```diff
@@ -65,6 +77,33 @@ func makeTrackerRequest(remoteAddr, remoteHost, remotePort string, createTime ti
 	}
 }
 
+// remoteAddress a best guessed remote address for the request.
+// Rational is the following:
+// remoteAddress() is used either for logging or as rootURL for creating a new wsPeer.
+// rootURL is an address to connect to. It is well defined only for peers from a phonebooks,
+// and for incoming peers the best guess is either otherPublicAddr, remoteHost, or remoteAddr.
+//   - otherPublicAddr is provided by a remote peer by X-Algorand-Location header and cannot be trusted,
+//     but can be used if remoteHost matches to otherPublicAddr value. In this case otherPublicAddr is a better guess
+//     for a rootURL because it might include a port.
+//   - remoteHost is either a real address of the remote peer or a value of X-Forwarded-For header.
+//     Use it if remoteHost was taken from X-Forwarded-For header.
+//     Note, the remoteHost does not include a port since a listening port is not known.
+//   - remoteAddr is used otherwise.
+func (tr *TrackerRequest) remoteAddress() string {
+	if len(tr.otherPublicAddr) != 0 {
+		url, err := ParseHostOrURL(tr.otherPublicAddr)
+		if err == nil && len(tr.remoteHost) > 0 && url.Hostname() == tr.remoteHost {
+			return tr.otherPublicAddr
+		}
+	}
+	url, err := ParseHostOrURL(tr.remoteAddr)
+	if err != nil || url.Hostname() == tr.remoteHost {
```

Comment:
Am I reading this right or is the remoteHost and remoteAddr in the description the inverse of how they are used in this section?

Reply:
The idea is the following: `remoteAddr` is source of truth and a default value. The `url.Hostname() == tr.remoteHost` handles the case with no proxy and `tr.remoteAddr` has the port number, and `remoteHost` does not.

Reply:
updated